src/Helpers/JWKFetcher.php

```diff
@@ -59,101 +62,135 @@ protected function convertCertToPem($cert)
         return $output;
     }
 
+    // phpcs:disable
     /**
-     * Fetch x509 cert for RS256 token decoding.
+     * TODO: Deprecate, use $this->getJwksX5c() instead.
+     *
+     * @param string $iss
+     *
+     * @return array|mixed|null
      *
-     * @param string      $domain Base domain for the JWKS, including scheme.
-     * @param string|null $kid    Kid to use.
-     * @param string      $path   Path to the JWKS from the $domain.
+     * @throws \Exception
      *
-     * @return mixed
      *
-     * @throws \Exception If the Guzzle HTTP client cannot complete the request.
+     * @codeCoverageIgnore
      */
-    public function fetchKeys($domain, $kid = null, $path = '.well-known/jwks.json')
+    public function fetchKeys($iss)
     {
-        $jwks_url = $domain.$path;
-
-        // Check for a cached JWKS value.
-        $secret = $this->cache->get($jwks_url);
-        if (! is_null($secret)) {
-            return $secret;
-        }
-
-        $secret = [];
-
-        $jwks = $this->getJwks( $domain, $path );
-
-        if (! is_array( $jwks['keys'] ) || empty( $jwks['keys'] )) {
-            return $secret;
-        }
-
-        // No kid passed so get the kid of the first JWK.
-        if (is_null($kid)) {
-            $kid = $this->getProp( $jwks, 'kid' );
-        }
-
-        $x5c = $this->getProp( $jwks, 'x5c', $kid );
+        $url = "{$iss}.well-known/jwks.json";
+        if (($secret = $this->cache->get($url)) === null) {
+            $secret  = [];
+            $request = new RequestBuilder([
+                'domain' => $iss,
+                'basePath' => '.well-known/jwks.json',
+                'method' => 'GET',
+                'guzzleOptions' => $this->guzzleOptions
+            ]);
+            $jwks    = $request->call();
+            foreach ($jwks['keys'] as $key) {
+                $secret[$key['kid']] = $this->convertCertToPem($key['x5c'][0]);
+            }
 
-        // Need the kid and x5c for a well-formed return value. 
-        if (! is_null($kid) && ! is_null($x5c)) {
-            $secret[$kid] = $this->convertCertToPem($x5c);
-            $this->cache->set($jwks_url, $secret);
+            $this->cache->set($url, $secret);
         }
 
         return $secret;
     }
+    // phpcs:enable
 
     /**
-     * Get a specific property from a JWKS using a key, if provided.
+     * Fetch x509 cert for RS256 token decoding.
      *
-     * @param array       $jwks JWKS to parse.
-     * @param string      $prop Property to retrieve.
-     * @param null|string $kid  Kid to check.
+     * @param string      $jwks_url URL to the JWKS.
+     * @param string|null $kid      Key ID.
```

Comment:
"if not provided and a single JSON Web Key exists on the set, that one will be returned"